R/undesirable_function_linter.R

```diff
@@ -0,0 +1,58 @@
+default_undesirable_funs <- c(
+  "setwd"="use withr::with_dir()",
+  "sink"="use withr::with_sink()",
+  "par"="use withr::with_par()",
+  "options"="use withr::with_options()",
+  "Sys.setenv"="use withr::with_envvar()",
+  "Sys.setlocale"="use withr::with_locale()",
+  ".libPaths"="use withr::with_libpaths()",
+  "attach"="use roxygen2's @importFrom statement in packages, or `::` in scripts",
+  "detach"="use roxygen2's @importFrom statement in packages, or `::` in scripts",
+  "library"= "use roxygen2's @importFrom statement in packages, or `::` in scripts",
+  "require"= "use roxygen2's @importFrom statement in packages, or `::` in scripts",
+  "loadNamespace"="use `::` or requireNamespace()",
```

Comment:
If you are generally aiming to follow Hadley's style and guidelines, `loadNamespace` should be left in by default.